controller/controller.js

```diff
@@ -27,5 +27,11 @@ exports.getArticleById = (req, res, next) => {
     .then((article) => {
       res.status(200).send({ article });
     })
-    .catch(next);
+    .catch((err) => {
+      if (err.code === "22P02") {
```

Comment:
Almost Musab! This logic should be handled by the error handler function specifically for PSQL errors. Have a look at NC Notes - Back End - Error Handling - PSQL Thrown Errors for more guidance on how to structure this